### PR TITLE
fix(http): improve error logging

### PR DESCRIPTION
**Pull request by rarkins**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/npm/get.ts [hunk 1]
```diff
@@ -157,35 +157,12 @@ export async function getDependency(
...
-        { err, packageName },
-        `Dependency lookup failure: not found`
-      );
+    const ignoredStatusCodes = [401, 402, 403, 404];
+    const ignoredResponseCodes = ['ENOTFOUND'];
+    if (
+      ignoredStatusCodes.includes(err.statusCode) ||
+      ignoredResponseCodes.includes(err.code)
+    ) {
       return null;
```
Comment (rarkins) [inline lib/modules/datasource/npm/get.ts:166]: These will already be logged higher up, and the additional messages here don't benefit

File: lib/util/http/github.ts [hunk 2]
```diff
@@ -154,11 +154,6 @@ function handleGotError(
   ) {
     return err;
   }
-  if (err.statusCode === 404) {
-    logger.debug({ url: path }, 'GitHub 404');
-  } else {
-    logger.debug({ err }, 'Unknown GitHub error');
-  }
```
Comment (rarkins) [inline lib/util/http/github.ts:161]: These will be logged lower down (debug in http layer) and higher up (lookup error)

File: lib/config/presets/local/common.ts [hunk 3]
```diff
@@ -17,10 +17,7 @@ export async function fetchJSONFile(
       throw err;
     }
 
-    logger.debug(
-      { err, repo, fileName },
-      `Failed to retrieve ${fileName} from repo ${repo}`
-    );
+    logger.debug(`Could not fetch ${fileName} from repo ${repo}`);
```
Comment (viceice) [inline lib/config/presets/local/common.ts:20]: why not same message as the others?
```suggestion
    logger.debug(`Preset file ${fileName} not found in ${repo}`);
```

**Review by viceice — COMMENTED** [review]
otherwise LGTM

**Review by viceice — APPROVED** [review]

**Pull request merged**